tricore warnings

```
arch/TriCore/TriCoreDisassembler.c:188:6: warning: no previous prototype for function 'TriCore_getFeatureBits' [-Wmissing-prototypes]
bool TriCore_getFeatureBits(unsigned int mode, unsigned int feature)
     ^
arch/TriCore/TriCoreDisassembler.c:188:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
bool TriCore_getFeatureBits(unsigned int mode, unsigned int feature)
^
static
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.3/include/stdbool.h:22:14: note: expanded from macro 'bool'
#define bool _Bool
             ^
arch/TriCore/TriCoreInstPrinter.c:517:6: warning: no previous prototype for function 'set_mem_access' [-Wmissing-prototypes]
void set_mem_access(MCInst *MI, unsigned int access)
     ^
arch/TriCore/TriCoreInstPrinter.c:517:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
void set_mem_access(MCInst *MI, unsigned int access)
^
static
In file included from arch/TriCore/TriCoreInstPrinter.c:524:
arch/TriCore/TriCoreGenAsmWriter.inc:19:18: warning: no previous prototype for function 'getMnemonic' [-Wmissing-prototypes]
MnemonicBitsInfo getMnemonic(MCInst *MI, SStream *O) {
                 ^
arch/TriCore/TriCoreGenAsmWriter.inc:19:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
MnemonicBitsInfo getMnemonic(MCInst *MI, SStream *O) {
^
static
arch/TriCore/TriCoreGenAsmWriter.inc:3685:6: warning: no previous prototype for function 'printAliasInstr' [-Wmissing-prototypes]
bool printAliasInstr(MCInst *MI, uint64_t Address, SStream *OS) {
     ^
arch/TriCore/TriCoreGenAsmWriter.inc:3685:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
bool printAliasInstr(MCInst *MI, uint64_t Address, SStream *OS) {
^
static
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.3/include/stdbool.h:22:14: note: expanded from macro 'bool'
#define bool _Bool
             ^
arch/TriCore/TriCoreModule.c:11:8: warning: no previous prototype for function 'TRICORE_global_init' [-Wmissing-prototypes]
cs_err TRICORE_global_init(cs_struct *ud)
       ^
arch/TriCore/TriCoreModule.c:11:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
cs_err TRICORE_global_init(cs_struct *ud)
^
static
arch/TriCore/TriCoreModule.c:32:8: warning: no previous prototype for function 'TRICORE_option' [-Wmissing-prototypes]
cs_err TRICORE_option(cs_struct *handle, cs_opt_type type, size_t value)
       ^
arch/TriCore/TriCoreModule.c:32:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
cs_err TRICORE_option(cs_struct *handle, cs_opt_type type, size_t value)
^
static
2 warnings generated.
arch/TriCore/TriCoreDisassembler.c:1521:6: warning: no previous prototype for function 'TriCore_getInstruction' [-Wmissing-prototypes]
bool TriCore_getInstruction(csh ud, const uint8_t *code, size_t code_len,
     ^
arch/TriCore/TriCoreDisassembler.c:1521:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
bool TriCore_getInstruction(csh ud, const uint8_t *code, size_t code_len,
^
static
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.3/include/stdbool.h:22:14: note: expanded from macro 'bool'
#define bool _Bool
             ^
arch/TriCore/TriCoreDisassembler.c:1570:6: warning: no previous prototype for function 'TriCore_init' [-Wmissing-prototypes]
void TriCore_init(MCRegisterInfo *MRI)
     ^
arch/TriCore/TriCoreDisassembler.c:1570:1: note: declare 'static' if the function is not intended to be used outside of this translation unit
void TriCore_init(MCRegisterInfo *MRI)
^
static
```

@trufae What compiler flags are you using? I don't see the compilation warnings you mentioned, when I use `-DCMAKE_C_FLAGS="-Wall -Wextra" -DCMAKE_CXX_FLAGS="-Wall -Wextra"` and **clang14** to compile **capstone@next**, 

```
λ ninja -C cmake-build-debug-clang-wall-wextra/ | rg -i tricore
[67/152] Building C object CMakeFiles/capstone.dir/arch/TriCore/TriCoreModule.c.o
[68/152] Building C object CMakeFiles/capstone.dir/arch/TriCore/TriCoreInstPrinter.c.o
/home/aya/Source/capstone/arch/TriCore/TriCoreInstPrinter.c:38:31: warning: unused parameter 'ud' [-Wunused-parameter]
void TriCore_post_printer(csh ud, cs_insn *insn, char *insn_asm, MCInst *mci)
/home/aya/Source/capstone/arch/TriCore/TriCoreInstPrinter.c:38:44: warning: unused parameter 'insn' [-Wunused-parameter]
void TriCore_post_printer(csh ud, cs_insn *insn, char *insn_asm, MCInst *mci)
/home/aya/Source/capstone/arch/TriCore/TriCoreInstPrinter.c:38:56: warning: unused parameter 'insn_asm' [-Wunused-parameter]
void TriCore_post_printer(csh ud, cs_insn *insn, char *insn_asm, MCInst *mci)
/home/aya/Source/capstone/arch/TriCore/TriCoreInstPrinter.c:38:74: warning: unused parameter 'mci' [-Wunused-parameter]
void TriCore_post_printer(csh ud, cs_insn *insn, char *insn_asm, MCInst *mci)
/home/aya/Source/capstone/arch/TriCore/TriCoreInstPrinter.c:286:12: warning: comparison of integers of different signs: 'uint64_t' (aka 'unsigned long') and 'int' [-Wsign-compare]
/home/aya/Source/capstone/arch/TriCore/TriCoreInstPrinter.c:517:29: warning: unused parameter 'MI' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreInstPrinter.c:517:46: warning: unused parameter 'access' [-Wunused-parameter]
In file included from /home/aya/Source/capstone/arch/TriCore/TriCoreInstPrinter.c:524:
/home/aya/Source/capstone/arch/TriCore/TriCoreGenAsmWriter.inc:19:51: warning: unused parameter 'O' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreGenAsmWriter.inc:3011:44: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreGenAsmWriter.inc:3685:30: warning: unused parameter 'MI' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreGenAsmWriter.inc:3685:43: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreGenAsmWriter.inc:3685:61: warning: unused parameter 'OS' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreInstPrinter.c:526:41: warning: unused parameter 'handle' [-Wunused-parameter]
const char *TriCore_getRegisterName(csh handle, unsigned int id)
/home/aya/Source/capstone/arch/TriCore/TriCoreInstPrinter.c:535:54: warning: unused parameter 'Info' [-Wunused-parameter]
void TriCore_printInst(MCInst *MI, SStream *O, void *Info)
[78/152] Building C object CMakeFiles/capstone.dir/arch/TriCore/TriCoreMapping.c.o
/home/aya/Source/capstone/arch/TriCore/TriCoreMapping.c:93:35: warning: unused parameter 'handle' [-Wunused-parameter]
const char *TriCore_insn_name(csh handle, unsigned int id)
/home/aya/Source/capstone/arch/TriCore/TriCoreMapping.c:121:36: warning: unused parameter 'handle' [-Wunused-parameter]
const char *TriCore_group_name(csh handle, unsigned int id)
[81/152] Building C object CMakeFiles/capstone.dir/arch/TriCore/TriCoreDisassembler.c.o
In file included from /home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:217:
/home/aya/Source/capstone/arch/TriCore/TriCoreGenDisassemblerTables.inc:4040:1: warning: unused parameter 'feature' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreGenDisassemblerTables.inc:3935:70: note: expanded from macro 'DecodeInstruction'
/home/aya/Source/capstone/arch/TriCore/TriCoreGenDisassemblerTables.inc:4044:1: warning: unused parameter 'feature' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreGenDisassemblerTables.inc:3935:70: note: expanded from macro 'DecodeInstruction'
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:255:15: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:255:30: warning: unused parameter 'Decoder' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:270:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:292:15: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:292:30: warning: unused parameter 'Decoder' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:307:15: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:335:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:359:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:384:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:425:21: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:425:36: warning: unused parameter 'Decoder' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:443:15: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:497:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:563:15: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:597:17: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:635:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:682:15: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:744:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:776:17: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:798:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:822:17: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:844:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:866:17: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:895:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:895:31: warning: unused parameter 'Decoder' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:918:17: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:918:32: warning: unused parameter 'Decoder' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:935:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:960:17: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:983:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1004:17: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1040:17: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1080:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1118:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1152:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1186:17: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1226:17: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1260:17: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1299:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1342:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1368:16: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1405:17: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1405:32: warning: unused parameter 'Decoder' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1427:17: warning: unused parameter 'Address' [-Wunused-parameter]
/home/aya/Source/capstone/arch/TriCore/TriCoreDisassembler.c:1448:16: warning: unused parameter 'Address' [-Wunused-parameter]
[96/152] Building C object CMakeFiles/test_tricore.dir/tests/test_tricore.c.o
[115/152] Building C object CMakeFiles/cstool.dir/cstool/cstool_tricore.c.o
[146/152] Linking C executable test_tricore
```

--- a/arch/TriCore/TriCoreDisassembler.c
+++ b/arch/TriCore/TriCoreDisassembler.c
@@ -26,6 +26,8 @@
 #include "../../MCDisassembler.h"
 #include "../../MathExtras.h"
 
+#include "TriCoreDisassembler.h"
+
 static bool readInstruction16(const uint8_t *code, size_t code_len,
 			      uint16_t *insn)
 {

--- a/arch/TriCore/TriCoreDisassembler.h
+++ b/arch/TriCore/TriCoreDisassembler.h
@@ -18,4 +18,6 @@ bool TriCore_getInstruction(csh ud, const uint8_t *code, size_t code_len,
 			    MCInst *instr, uint16_t *size, uint64_t address,
 			    void *info);
 
+bool TriCore_getFeatureBits(unsigned int mode, unsigned int feature);
+
 #endif

--- a/arch/TriCore/TriCoreInstPrinter.c
+++ b/arch/TriCore/TriCoreInstPrinter.c
@@ -514,10 +514,13 @@ typedef struct {
 	uint64_t second; // Bits
 } MnemonicBitsInfo;
 
-void set_mem_access(MCInst *MI, unsigned int access)
+static void set_mem_access(MCInst *MI, unsigned int access)
 {
 	// TODO: TriCore
 }
+
+MnemonicBitsInfo getMnemonic(MCInst *MI, SStream *O);
+bool printAliasInstr(MCInst *MI, uint64_t Address, SStream *OS);
 
 #define PRINT_ALIAS_INSTR
 

--- a/arch/TriCore/TriCoreMapping.h
+++ b/arch/TriCore/TriCoreMapping.h
@@ -15,4 +15,7 @@ const char *TriCore_insn_name(csh handle, unsigned int id);
 
 const char *TriCore_group_name(csh handle, unsigned int id);
 
+cs_err TRICORE_global_init(cs_struct *ud);
+cs_err TRICORE_option(cs_struct *handle, cs_opt_type type, size_t value);
+
 #endif

--- a/cstool/cstool.c
+++ b/cstool/cstool.c
@@ -7,6 +7,7 @@
 #include "getopt.h"
 
 #include <capstone/capstone.h>
+#include "cstool.h"
 
 void print_string_hex(const char *comment, unsigned char *str, size_t len);
 
@@ -114,25 +115,6 @@ static struct {
 	{ "tc162", CS_ARCH_TRICORE, CS_MODE_TRICORE_162 },
 	{ NULL }
 };
-
-void print_insn_detail_x86(csh ud, cs_mode mode, cs_insn *ins);
-void print_insn_detail_arm(csh handle, cs_insn *ins);
-void print_insn_detail_arm64(csh handle, cs_insn *ins);
-void print_insn_detail_mips(csh handle, cs_insn *ins);
-void print_insn_detail_ppc(csh handle, cs_insn *ins);
-void print_insn_detail_sparc(csh handle, cs_insn *ins);
-void print_insn_detail_sysz(csh handle, cs_insn *ins);
-void print_insn_detail_xcore(csh handle, cs_insn *ins);
-void print_insn_detail_m68k(csh handle, cs_insn *ins);
-void print_insn_detail_tms320c64x(csh handle, cs_insn *ins);
-void print_insn_detail_m680x(csh handle, cs_insn *ins);
-void print_insn_detail_evm(csh handle, cs_insn *ins);
-void print_insn_detail_riscv(csh handle, cs_insn *ins);
-void print_insn_detail_wasm(csh handle, cs_insn *ins);
-void print_insn_detail_mos65xx(csh handle, cs_insn *ins);
-void print_insn_detail_bpf(csh handle, cs_insn *ins);
-void print_insn_detail_sh(csh handle, cs_insn *ins);
-void print_insn_detail_tricore(csh handle, cs_insn *ins);
 
 static void print_details(csh handle, cs_arch arch, cs_mode md, cs_insn *ins);
 

--- a/cstool/cstool.h
+++ b/cstool/cstool.h
@@ -1,0 +1,23 @@
+#ifndef CAPSTONE_CSTOOL_CSTOOL_H_
+#define CAPSTONE_CSTOOL_CSTOOL_H_
+
+void print_insn_detail_x86(csh ud, cs_mode mode, cs_insn *ins);
+void print_insn_detail_arm(csh handle, cs_insn *ins);
+void print_insn_detail_arm64(csh handle, cs_insn *ins);
+void print_insn_detail_mips(csh handle, cs_insn *ins);
+void print_insn_detail_ppc(csh handle, cs_insn *ins);
+void print_insn_detail_sparc(csh handle, cs_insn *ins);
+void print_insn_detail_sysz(csh handle, cs_insn *ins);
+void print_insn_detail_xcore(csh handle, cs_insn *ins);
+void print_insn_detail_m68k(csh handle, cs_insn *ins);
+void print_insn_detail_tms320c64x(csh handle, cs_insn *ins);
+void print_insn_detail_m680x(csh handle, cs_insn *ins);
+void print_insn_detail_evm(csh handle, cs_insn *ins);
+void print_insn_detail_riscv(csh handle, cs_insn *ins);
+void print_insn_detail_wasm(csh handle, cs_insn *ins);
+void print_insn_detail_mos65xx(csh handle, cs_insn *ins);
+void print_insn_detail_bpf(csh handle, cs_insn *ins);
+void print_insn_detail_sh(csh handle, cs_insn *ins);
+void print_insn_detail_tricore(csh handle, cs_insn *ins);
+
+#endif //CAPSTONE_CSTOOL_CSTOOL_H_

--- a/cstool/cstool_arm.c
+++ b/cstool/cstool_arm.c
@@ -2,8 +2,7 @@
 #include <stdlib.h>
 
 #include <capstone/capstone.h>
-
-void print_insn_detail_arm(csh handle, cs_insn *ins);
+#include "cstool.h"
 
 void print_insn_detail_arm(csh handle, cs_insn *ins)
 {

--- a/cstool/cstool_arm64.c
+++ b/cstool/cstool_arm64.c
@@ -5,8 +5,7 @@
 #include <stdlib.h>
 
 #include <capstone/capstone.h>
-
-void print_insn_detail_arm64(csh handle, cs_insn *ins);
+#include "cstool.h"
 
 void print_insn_detail_arm64(csh handle, cs_insn *ins)
 {

--- a/cstool/cstool_bpf.c
+++ b/cstool/cstool_bpf.c
@@ -2,12 +2,11 @@
 
 #include <capstone/capstone.h>
 #include <capstone/platform.h>
+#include "cstool.h"
 
 static const char * ext_name[] = {
 	[BPF_EXT_LEN] = "#len",
 };
-
-void print_insn_detail_bpf(csh handle, cs_insn *ins);
 
 void print_insn_detail_bpf(csh handle, cs_insn *ins)
 {

--- a/cstool/cstool_evm.c
+++ b/cstool/cstool_evm.c
@@ -2,8 +2,7 @@
 #include <stdlib.h>
 
 #include <capstone/capstone.h>
-
-void print_insn_detail_evm(csh handle, cs_insn *ins);
+#include "cstool.h"
 
 void print_insn_detail_evm(csh handle, cs_insn *ins)
 {

--- a/cstool/cstool_m680x.c
+++ b/cstool/cstool_m680x.c
@@ -3,13 +3,11 @@
 
 #include <stdio.h>
 #include <capstone/capstone.h>
-
-void print_insn_detail_m680x(csh handle, cs_insn *insn);
+#include "cstool.h"
 
 static const char *s_access[] = {
 	"UNCHANGED", "READ", "WRITE", "READ | WRITE",
 };
-
 
 static void print_read_write_regs(csh handle, cs_detail *detail)
 {

--- a/cstool/cstool_m68k.c
+++ b/cstool/cstool_m68k.c
@@ -8,8 +8,7 @@
 
 #include <stdio.h>
 #include <capstone/capstone.h>
-
-void print_insn_detail_m68k(csh handle, cs_insn *ins);
+#include "cstool.h"
 
 static const char* s_addressing_modes[] = {
 	"<invalid mode>",

--- a/cstool/cstool_mips.c
+++ b/cstool/cstool_mips.c
@@ -5,8 +5,7 @@
 #include <stdlib.h>
 
 #include <capstone/capstone.h>
-
-void print_insn_detail_mips(csh handle, cs_insn *ins);
+#include "cstool.h"
 
 void print_insn_detail_mips(csh handle, cs_insn *ins)
 {

--- a/cstool/cstool_mos65xx.c
+++ b/cstool/cstool_mos65xx.c
@@ -2,9 +2,7 @@
 #include <stdlib.h>
 
 #include <capstone/capstone.h>
-
-void print_insn_detail_mos65xx(csh handle, cs_insn *ins);
-
+#include "cstool.h"
 
 static const char *get_am_name(mos65xx_address_mode mode)
 {
@@ -64,7 +62,6 @@ static const char *get_am_name(mos65xx_address_mode mode)
 			return "stack relative indirect indexed with y";
 	}
 }
-
 
 void print_insn_detail_mos65xx(csh handle, cs_insn *ins)
 {

--- a/cstool/cstool_powerpc.c
+++ b/cstool/cstool_powerpc.c
@@ -4,9 +4,7 @@
 #include <stdio.h>
 
 #include <capstone/capstone.h>
-
-void print_insn_detail_ppc(csh handle, cs_insn *ins);
-
+#include "cstool.h"
 
 static const char* get_bc_name(int bc)
 {

--- a/cstool/cstool_riscv.c
+++ b/cstool/cstool_riscv.c
@@ -3,8 +3,7 @@
 
 #include <stdio.h>
 #include <capstone/capstone.h>
-
-void print_insn_detail_riscv(csh handle, cs_insn *ins);
+#include "cstool.h"
 
 void print_insn_detail_riscv(csh handle, cs_insn *ins)
 {

--- a/cstool/cstool_sh.c
+++ b/cstool/cstool_sh.c
@@ -6,8 +6,7 @@
 
 #include <stdio.h>
 #include <capstone/capstone.h>
-
-void print_insn_detail_sh(csh handle, cs_insn *ins);
+#include "cstool.h"
 
 static const char* s_addressing_modes[] = {
 	"<invalid mode>",

--- a/cstool/cstool_sparc.c
+++ b/cstool/cstool_sparc.c
@@ -4,9 +4,7 @@
 #include <stdio.h>
 
 #include <capstone/capstone.h>
-
-void print_insn_detail_sparc(csh handle, cs_insn *ins);
-
+#include "cstool.h"
 
 void print_insn_detail_sparc(csh handle, cs_insn *ins)
 {

--- a/cstool/cstool_systemz.c
+++ b/cstool/cstool_systemz.c
@@ -4,8 +4,7 @@
 #include <stdio.h>
 
 #include <capstone/capstone.h>
-
-void print_insn_detail_sysz(csh handle, cs_insn *ins);
+#include "cstool.h"
 
 void print_insn_detail_sysz(csh handle, cs_insn *ins)
 {

--- a/cstool/cstool_tms320c64x.c
+++ b/cstool/cstool_tms320c64x.c
@@ -3,8 +3,7 @@
 
 #include <stdio.h>
 #include <capstone/capstone.h>
-
-void print_insn_detail_tms320c64x(csh handle, cs_insn *ins);
+#include "cstool.h"
 
 void print_insn_detail_tms320c64x(csh handle, cs_insn *ins)
 {

--- a/cstool/cstool_tricore.c
+++ b/cstool/cstool_tricore.c
@@ -2,7 +2,6 @@
 #include <stdlib.h>
 
 #include <capstone/capstone.h>
-
 #include "cstool.h"
 
 void print_insn_detail_tricore(csh handle, cs_insn *ins)

--- a/cstool/cstool_tricore.c
+++ b/cstool/cstool_tricore.c
@@ -3,6 +3,8 @@
 
 #include <capstone/capstone.h>
 
+#include "cstool.h"
+
 void print_insn_detail_tricore(csh handle, cs_insn *ins)
 {
 	cs_tricore *tricore;

--- a/cstool/cstool_wasm.c
+++ b/cstool/cstool_wasm.c
@@ -2,8 +2,7 @@
 #include <stdlib.h>
 
 #include <capstone/capstone.h>
-
-void print_insn_detail_wasm(csh handle, cs_insn *ins);
+#include "cstool.h"
 
 void print_insn_detail_wasm(csh handle, cs_insn *ins)
 {

--- a/cstool/cstool_x86.c
+++ b/cstool/cstool_x86.c
@@ -4,8 +4,7 @@
 #include <stdlib.h>
 
 #include <capstone/capstone.h>
-
-void print_insn_detail_x86(csh ud, cs_mode mode, cs_insn *ins);
+#include "cstool.h"
 
 void print_string_hex(const char *comment, unsigned char *str, size_t len);
 

--- a/cstool/cstool_xcore.c
+++ b/cstool/cstool_xcore.c
@@ -3,8 +3,7 @@
 
 #include <stdio.h>
 #include <capstone/capstone.h>
-
-void print_insn_detail_xcore(csh handle, cs_insn *ins);
+#include "cstool.h"
 
 void print_insn_detail_xcore(csh handle, cs_insn *ins)
 {

